### PR TITLE
Added new recipe for Traversal Induced Values.

### PR DESCRIPTION
**Pull request by spmallette**
Just a documentation addition. I learned that it's not so good to CTR these as there is sometimes good feedback on Gremlin code that should be incorporated and a PR is the best way to go about gathering that feedback.

VOTE +1

### CODE REVIEW

**Comment by spmallette** [comment]
Please ignore the failing test:

 IdentityRemovalStrategyTest$PerformanceTest>TraversalStrategyPerformanceTest.shouldBeFaster:107 null

that's an issue raised elsewhere: https://issues.apache.org/jira/browse/TINKERPOP-1390

**Comment by twilmes** [comment]
I might be missing something silly here, but It looks like you need to commit `traversal-induced-values.asciidoc`.  I'm only seeing the updates to the changelog and index on this branch.

**Comment by spmallette** [comment]
Sorry - forgot to add the file @twilmes - fixed up the commit and force pushed.

**Comment by twilmes** [comment]
Looks good, nice choice for a recipe.

VOTE: +1

**Comment by okram** [comment]
VOTE +1.

This got me to thinking that we really need to start to extend other steps (beyond the `Mutating`) to support traversal parameterization.

File: docs/src/recipes/traversal-induced-values.asciidoc [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+<1> Find the "marko" `Vertex` and label it as "marko".
+<2> Traverse out on the "knows" edges to the adjacent `Vertex` and label it as "person".
+<3> Filter the incoming "person" vertices. It is within this filter, that the traversal induced values are utilized.
+The inner `select` grabs the "marko" vertex and the current "friend". The `by` modulator extracts the "age" from both
+of those vertices which yields a `Map` with two keys, "marko" and "friend", where the value of each is the "age".
+<4> The `Map` produced in the previous step can then be filtered with `where` to only return a result if the "friend"
+age is greater than the "marko" age. If this is successful, then the `filter` step from the previous line will succeed
+and allow the "friend" vertex to pass through.
+
+This traversal could also be written declaratively with `match` step as follows:
```
Comment (pluradj) [inline docs/src/recipes/traversal-induced-values.asciidoc]: Should there be any discussion of the possible performance differences between the 2 approaches? Or is that mostly a system-specific discussion?

Comment (spmallette) [inline docs/src/recipes/traversal-induced-values.asciidoc]: I thought about going into more detail there, but opted not to. I kinda thought that if we wanted that sort of content that it didn't belong in a recipe but in reference docs. It seemed a little out of scope of the recipe.

**Pull request merged**